app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SessionsController < Devise::SessionsController
+  def new
+    super
+  end
+
+  def create
+    if(current_company == nil)
+      flash[:danger] = 'Invalid Company!'
+      return redirect_to new_employee_session_path
+    end
+    employee = Employee.find_by(email: params[:employee][:email], company_id: current_company.id)
+    if employee == nil || !employee.valid_password?(params[:employee][:password])
```

Comment:
blank? and same thing here

Reply:
same answer, invalid password will appear. Tested as well

Reply:
use employee.blank? instead of employee == nil